xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-war/src/main/webapp/resources/uicomponents/widgets/tree.js

```diff
@@ -38,6 +38,16 @@ require([paths.treeRequireConfig], function() {
         // Open the tree to the specified node and select it.
         openToNodeId && data.instance.openTo(openToNodeId);
       });
+
+      $('.xtree.dbTreeList').on('changed.jstree', function(event, data) {
+        if (typeof data.instance.settings.xwiki.fieldName !== 'undefined') {
+          var fieldName = data.instance.settings.xwiki.fieldName;
+          var fieldValue = data.selected.join('|');
+          $('input[type="hidden"]').filter(function() {
+            return $(this).attr('name') === fieldName;
+          }).val(fieldValue);
+        }
+      });
```

Comment:
I'd rather move this code to the ``tree.js`` inside the ``xwiki-platform-tree-webjar`` because the ``tree.js`` from ``xwiki-platform-tree-war`` is just a (**temporary**) loader until we implement https://jira.xwiki.org/browse/XWIKI-12788 .

Reply:
Done